src/Core/Managed/Shared/Extensibility/Implementation/Tracing/CoreEventSource.cs

```diff
@@ -312,6 +312,42 @@ public void FailedToSend(string msg, string appDomainName = "Incorrect")
                 this.nameProvider.Name);
         }
 
+        [Event(
+            25,
+            Message = "Failed to flush aggregated metrics. Exception: {0}.",
+            Level = EventLevel.Error)]
+        public void FailedToFlushMetricAggregators(string ex, string appDomainName = "Incorrect")
+        {
+            this.WriteEvent(
+                25,
+                ex ?? string.Empty,
+                this.nameProvider.Name);
+        }
+
+        [Event(
+            26,
+            Message = "Failed to snapshot aggregated metrics. Exception: {0}.",
+            Level = EventLevel.Error)]
+        public void FailedToSnapshotMetricAggregators(string ex, string appDomainName = "Incorrect")
+        {
+            this.WriteEvent(
+                26,
+                ex ?? string.Empty,
+                this.nameProvider.Name);
+        }
+
+        [Event(
+            27,
+            Message = "Failed to invoke metric processor. Exception: {0}.",
+            Level = EventLevel.Error)]
```

Comment:
Level "Error" means that this trace will be sent to the portal as SDK "self-diagnostics" event. To make it more understandable for the user I'd recommend to mark it "User Actionable", trace the processor name and write instruction that "if issue persists - disable this processor"